### PR TITLE
Add count and index to collections

### PR DESCRIPTION
**Pull request by gnarf**

### CODE REVIEW

File: src/generators.ts [hunk 1]
```diff
@@ -62,3 +62,36 @@ export function last<T>(iter: Iterable<T>): T {
...
+    const { done: doneNow, value } = runner.next();
+    if (doneNow) return null;
+    return value;
+}
+
+/**
+ * Get count from iterable, using "length" property if it exists on the iterable.
+ * @param iterable Iterable or array to get length from
+ */
+export function count(iterable: Iterable<unknown> | unknown[]): number {
```
Comment (evmiguel) [inline src/generators.ts]: Why unknown and not T?

File: src/generators.ts [hunk 2]
```diff
@@ -62,3 +62,36 @@ export function last<T>(iter: Iterable<T>): T {
...
+
+export function getIndex<T>(iter: Iterable<T> | T[], index: number): T {
+    if (Array.isArray(iter)) {
+        return iter[index];
+    }
+
+    let counter = 0;
+    const runner = iter[Symbol.iterator]();
+    let done = false;
+    while (counter++ < index && !done) {
```
Comment (evmiguel) [inline src/generators.ts]: Sleek. I like this way of moving the index pointer

File: src/generators.ts [hunk 3]
```diff
@@ -62,3 +62,36 @@ export function last<T>(iter: Iterable<T>): T {
...
+        done = runner.next().done;
+    }
+    if (done) return null;
+    const { done: doneNow, value } = runner.next();
```
Comment (evmiguel) [inline src/generators.ts]: This makes sense, that the next() one is the element that you want

**Comment by evmiguel** [comment]
Non-blocking comments. Looks good to me

File: src/generators.ts (hunk 1 above)
Comment (gnarf) [reply src/generators.ts]: it can work on "any" but "unknown" is a not-lint-error way to specify "any" for utility methods like this

Comment (evmiguel) [reply src/generators.ts]: Nice!

**Pull request merged**